### PR TITLE
[bump] client: 2.0.0-rc.3.0.0 => 2.0.0-rc.3.0.1 (patch)

### PR DESCRIPTION
**Pull request by tylerbutler**
Bumped client from 2.0.0-rc.3.0.0 to 2.0.0-rc.3.0.1.

### CODE REVIEW

File: examples/benchmarks/bubblebench/simple-tree/package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
 		"@fluidframework/eslint-config-fluid": "^5.1.0",
 		"@fluidframework/test-tools": "^1.0.195075",
 		"@types/jest": "29.5.3",
-		"@types/jest-environment-puppeteer": "workspace:",
+		"@types/jest-environment-puppeteer": "workspace:~",
```
Comment (CraigMacomber) [inline examples/benchmarks/bubblebench/simple-tree/package.json:69]: this is odd. Any idea what's up with it?
Seems odd that we would have a workspace version of "@types/jest-environment-puppeteer" in the first place, and also odd that this PR would change how we depend on it.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1261,7 +1261,7 @@ importers:
       '@fluidframework/test-tools': ^1.0.195075
       '@fluidframework/tree': workspace:~
       '@types/jest': 29.5.3
-      '@types/jest-environment-puppeteer': 'workspace:'
+      '@types/jest-environment-puppeteer': workspace:~
```
Comment (CraigMacomber) [inline pnpm-lock.yaml:1264]: See https://github.com/microsoft/FluidFramework/pull/20685/files#r1567815513

File: examples/benchmarks/bubblebench/simple-tree/package.json (hunk 1 above)
Comment (tylerbutler) [reply examples/benchmarks/bubblebench/simple-tree/package.json:69]: There was a similar change in the 4.0 bump PR: https://github.com/microsoft/FluidFramework/pull/20661#discussion_r1567764270

I don't know the cause. But we definitely have an in-repo "copy" of the types package. See here:

https://github.com/microsoft/FluidFramework/blob/442cb76c288ca75919d71f3c0e1e3f5d4263ebbe/packages/test/types_jest-environment-puppeteer/index.d.ts#L6-L9

Comment (tylerbutler) [reply examples/benchmarks/bubblebench/simple-tree/package.json:69]: Sorry, I missed that this was package.json, not the lockfile. The old dependency was manually added, probably, and we normalize all the workspace deps as part of bumping. So this change and the corresponding lockfile change is correct.

**Review by CraigMacomber — APPROVED** [review]

**Pull request merged**